core/drag/drag-event.js

```diff
@@ -0,0 +1,371 @@
+var MutableEvent = require("../event/mutable-event").MutableEvent,
+    Montage = require("../core").Montage;
+
+// XXX Does not presently function server-side
+if (typeof window !== "undefined") {
+
+    var OBJECT_MINE_TYPE = 'application/object';
+
+    var DataTransfer = exports.DataTransfer = Montage.specialize({
+
+        __data: {
+            enumerable: false,
+            value: null
+        },
+
+        _data: {
+            enumerable: false,
+            get: function () {
+                return this.__data || (this.__data = new Map());
+            }
+        },
+
+        _dragImage: {
+            value: null,
+            enumerable: false,
+        },
+
+        _dragEffect: {
+            value: null
+        },
+
+        dragEffect: {
+            set: function (effect) {
+                if (DataTransfer.ALLOWED_EFFECTS.indexOf(effect) > -1) {
```

Comment:
We're repeating this: if (DataTransfer.ALLOWED_EFFECTS.indexOf(effect) > -1) quite a bit. I think it would make sense to create a method on DataTransfer like isAllowedEffect(effect) instead. And internally it might be faster to keep it as a table or object with properties rather than an array? For small amount of objects involved I'm not sure, but at least it would become an implementation detail.